### PR TITLE
doc: update readme with some technical info

### PR DESCRIPTION
**Pull request by joshuarli**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,6 +13,35 @@ bash install-devenv.sh
 **If you are setting up a new development environment, run `devenv bootstrap` after installing.**
 
 
+
+## what is it / technical overview
```
Comment (IanWoodard) [inline README.md:17]: Since this isn't a question, should we change it to: `what it is`?

File: README.md [hunk 2]
```diff
@@ -13,6 +13,35 @@ bash install-devenv.sh
 **If you are setting up a new development environment, run `devenv bootstrap` after installing.**
```
Comment (IanWoodard) [inline README.md:13]: I know this is unrelated to this PR, but I think we should explain that devenv bootstrap should be run in specific repos.

**Review by IanWoodard — APPROVED** [review]
A few nits, but otherwise lgtm

File: README.md (hunk 1 above)
Comment (joshuarli) [reply README.md:17]: i'll change it to just `technical overview`

File: README.md (hunk 2 above)
Comment (joshuarli) [reply README.md:13]: i'll change this to "If you are setting up a new laptop, run `devenv bootstrap` after installing."

it's a one-time thing for new engineers, not something that should be run in specific repos

**Review by IanWoodard — APPROVED** [review]

**Pull request merged**